modules/iam-types/src/main/scala/ch/epfl/bluebrain/nexus/commons/iam/types/auth/User.scala

```diff
@@ -0,0 +1,37 @@
+package ch.epfl.bluebrain.nexus.commons.iam.types.auth
+
+import ch.epfl.bluebrain.nexus.commons.iam.types.identity.Identity
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras.semiauto._
+import io.circe.{Decoder, Encoder}
+
+/**
+  * Base enumeration type for the internal user representation.
+  */
+sealed trait User extends Product with Serializable {
```

Comment:
could we merge `User` and `Caller` in a single type? the advice to introduce `Caller` was just for KG, I don't think we'll need it any more.